pkg/crypto_test.go

```diff
@@ -621,6 +617,88 @@ func TestCrypto_JWSSignEphemeralAndVerify(t *testing.T) {
 	})
 }
 
+func TestCrypto_GenerateVendorCACSR(t *testing.T) {
+	client := defaultBackend(t.Name())
+	defer emptyTemp(t.Name())
+
+	t.Run("ok", func(t *testing.T) {
```

Comment:
this is quite a length test, is it possible to split into smaller tests?

Reply:
Fixed by refactoring and adding subtests